libr/core/cmd_write.inc.c

```diff
@@ -2155,22 +2155,76 @@ static int cmd_wa(void *data, const char *input) {
 
 static int cmd_wb(void *data, const char *input) {
 	RCore *core = (RCore *)data;
-	ut8 b = core->block[0];
-	char *ui = r_str_newf ("%sb", r_str_trim_head_ro (input));
-	int uil = strlen (ui) - 1;
-	int n = r_num_get (NULL, ui);
-	free (ui);
-	if (uil > 8) {
-		R_LOG_ERROR ("wb only operates on bytes");
-	} else if (uil > 0) {
-		// Shift left and right to zero uil most significant bits
-		b <<= uil;
-		b >>= uil;
-		// Overwrite uil most significant bits and keep the rest
-		b |= (n << (8 - uil));
-		r_io_write_at (core->io, core->offset, &b, 1);
-	} else {
+	int uil = strlen (input);
+	char c;
+	int i;
+
+	// Check that user provided some input
+	if (uil == 0) {
 		r_core_cmd_help_match (core, help_msg_w, "wb");
+		return 0;
+	}
+
+	// Check that user input only contains binary data
+	for (i = 0; i < uil; i++) {
+		c = input[i];
+		// Ignore whitespaces, \r and \n chars
+		if (c == ' ' || c == '\r' || c == '\n') {
+			continue;
+		}
```

Comment:
`c == '\r'` is added due to windows. I don't have any windows machines to test on so I'm not sure if this is strictly necessary or not. Shouldn't hurt though.

Reply:
`isspace()` handles this for us.